pkg/apis/pipeline/v1/pipelineref_validation.go

```diff
@@ -47,8 +45,39 @@ func (ref *PipelineRef) Validate(ctx context.Context) (errs *apis.FieldError) {
 			}
 			errs = errs.Also(ValidateParameters(ctx, ref.Params))
 		}
-	} else if ref.Name == "" {
+		if ref.Resolver != "" {
```

Comment:
Could we "share" most of this between both `*ref_validation.go`  code ?

Reply:
Great idea! Fixed.